packages/contracts/tasks/deposits.ts

```diff
@@ -52,19 +70,51 @@ task('deposit', 'Deposits funds onto L2.')
     ).connect(l1Wallet)
 
     const amountWei = utils.parseEther(amountEth)
-    console.log(`Depositing ${amountEth} ETH to ${to}...`)
+    const value = amountWei.add(utils.parseEther('0.01'))
+    console.log(
+      `Depositing ${amountEth} ETH to ${to} with ${value.toString()} value...`
+    )
     // Below adds 0.01 ETH to account for gas.
     const tx = await depositFeed.depositTransaction(
       to,
       amountWei,
       '3000000',
       false,
       [],
-      {
-        value: amountWei.add(utils.parseEther('0.01')),
-      }
+      { value }
     )
     console.log(`Got TX hash ${tx.hash}. Waiting...`)
-    await tx.wait()
-    console.log('Done.')
+    const receipt = await tx.wait()
+
+    const event = receipt.events[0]
+    if (event?.event !== 'TransactionDeposited') {
+      throw new Error('Transaction not deposited')
+    }
+
+    let found = false
+    l2Provider.on('block', async (number) => {
```

Comment:
To make this a little bit more robust, we should handle the case where a new block with the deposit is produced on l2 immediately after the TransasctionDeposited event on l1.

Reply:
I agree with you that this is prone to race conditions, once https://github.com/ethereum-optimism/optimistic-specs/issues/348 is fixed then this will no longer be a problem and I'll remove this brute force check code

Reply:
Alright. Let's add a TODO here referencing the other issue so we don't forget.